components/net/fetch/methods.rs

```diff
@@ -240,6 +240,7 @@ pub fn main_fetch(
     let referrer_url = match mem::replace(&mut request.referrer, Referrer::NoReferrer) {
         Referrer::NoReferrer => None,
         Referrer::ReferrerUrl(referrer_source) | Referrer::Client(referrer_source) => {
+            request.headers.remove(header::REFERER);
```

Comment:
Was this change intended? Could you please add a spec number if possible?

Reply:
Oh, that was an existing step I'd earlier omitted by mistake. The step doesn't correspond to the spec, but I believe it removes Referer headers that were set in past redirects or preflights.

Reply:
Ok